src/components/AuthenticationPage.js

```diff
@@ -1,30 +1,97 @@
 import Greeting from "./Greeting";
 import Header from "./Header";
 import AuthenticationForm from "./AuthenticationForm";
+import axios from "axios";
 
-function AuthenticationPage({isSignin}){
-  const signup = {
-    title: "Sign Up",
-    linkMessage: "Have an account?",
-    link: "#",
-    placeholders: ["Username", "Email", "Password"],
-    buttonMessage: "Submit"
-  }
-  const signin = {
-    title: "Sign In",
-    linkMessage: "Need an account?",
-    link: "#",
-    placeholders: ["Email", "Password"],
-    buttonMessage: "Sign in"
-  }
+const signup = {
+  title: "Sign Up",
+  linkMessage: "Have an account?",
+  link: "#",
+  inputBox: [
+    { placeholder: "Username", name: "username", type: "text" },
+    { placeholder: "Email", name: "email", type: "email" },
+    { placeholder: "Password", name: "password", type: "password" },
+  ],
+  buttonMessage: "Submit",
+};
+const signin = {
+  title: "Sign In",
+  linkMessage: "Need an account?",
+  link: "#",
+  inputBox: [
+    { placeholder: "Email", name: "email", type: "email" },
+    { placeholder: "Password", name: "password", type: "password" },
+  ],
+  buttonMessage: "Sign in",
+};
+
+function AuthenticationPage({ isSignin }) {
   return (
     <div>
       <Header />
-      <div className="flex flex-col justify-center items-center">
-        <Greeting title={ isSignin ? signin.title : signup.title } linkMessage={ isSignin ? signin.linkMessage : signup.linkMessage } link={ isSignin ? signin.link : signup.link } />
-        <AuthenticationForm placeholders={ isSignin ? signin.placeholders : signup.placeholders } buttonMessage={ isSignin ? signin.buttonMessage : signup.buttonMessage } />
-      </div>
+      {isSignin ? (
+        <div className="flex flex-col justify-center items-center">
+          <Greeting
+            title={signin.title}
+            linkMessage={signin.linkMessage}
+            link={signin.link}
+          />
+          <AuthenticationForm
+            submitCallbackFn={handleSubmitSignIn}
+            inputBox={signin.inputBox}
+            buttonMessage={signin.buttonMessage}
+          />
+        </div>
+      ) : (
+        <div className="flex flex-col justify-center items-center">
+          <Greeting
+            title={signup.title}
+            linkMessage={signup.linkMessage}
+            link={signup.link}
+          />
+          <AuthenticationForm
+            submitCallbackFn={handleSubmitSignUp}
+            inputBox={signup.inputBox}
+            buttonMessage={signup.buttonMessage}
+          />
+        </div>
+      )}
     </div>
-  )
+  );
+}
+
+function handleSubmitSignIn() {}
+
+async function handleSubmitSignUp(event) {
+  try {
+    event.preventDefault();
+    const input = event.target;
+
+    const jwt = await axios({
+      method: "post",
+      url: "https://launchup-prisma.herokuapp.com/api/users",
```

Comment:
The base URL should be stored globally for the project and should not be hard coded. 
https://dev.to/amirsohel007/how-can-i-declare-a-global-variable-in-react-app-so-that-i-can-access-it-anywhere-in-the-application-2kjo/comments

Reply:
I used an environment variable for the backend API url .